### PR TITLE
test: validate expected nodes version during E2E

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

A PR in-flight has an odd issue w/ nodes not getting upgraded to the expected version. Unfortunately that didn't generate an E2E error.

This PR expects the api model k8s version will be reflected in the node version, for every node in every agent pool, including master nodes.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: test/e2e/kubernetes/node/node.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ type Address struct {
 type Info struct {
 	ContainerRuntimeVersion string `json:"containerRuntimeVersion"`
 	KubeProxyVersion        string `json:"kubeProxyVersion"`
-	KubeletProxyVersion     string `json:"kubeletVersion"`
```
Comment (jackfrancis) [inline test/e2e/kubernetes/node/node.go]: Not sure what happened here, it doesn't make sense for this property to be named `KubeletProxyVersion`, not to mention it deviates from the k8s type def (and it's not being used in the E2E code anywhere).

**Comment by CecileRobertMichon** [comment]
lgtm pending upgrade e2e test to validate this works

**Comment by mboersma** [comment]
/hold

pending confirmation from upgrade e2e test

**Comment by jackfrancis** [comment]
This needs more work as re-running E2E on an existing cluster after upgrade is not testing against the upgraded version, and so the upgrade test fails.

**Comment by jackfrancis** [comment]
validated upgrade

**Pull request merged**